src/libraries/Common/src/System/Collections/Generic/BitHelper.cs

```diff
@@ -19,19 +21,22 @@ internal BitHelper(Span<int> span, bool clear)
 
         internal void MarkBit(int bitPosition)
         {
-            int bitArrayIndex = bitPosition / IntSize;
-            if ((uint)bitArrayIndex < (uint)_span.Length)
+            (uint bitArrayIndex, uint position) = Math.DivRem((uint)bitPosition, IntSize);
+            Span<int> span = _span;
+            if (bitArrayIndex < (uint)span.Length)
             {
-                _span[bitArrayIndex] |= (1 << (bitPosition % IntSize));
+                span[(int)bitArrayIndex] |= 1 << (int)position;
```

Comment:
The bound-check is still elided, even when casted back to `int` here.
(good job from the JIT)